### PR TITLE
Make CT personality multi-tenant

### PR DESCRIPTION
**Pull request by daviddrysdale**
Ignore the "get inclusion proof for pre-cert" commit, as this is being reviewed under a separate [pull request](https://github.com/google/trillian/pull/267)

### CODE REVIEW

File: examples/ct/pem_cert_pool.go [hunk 1]
```diff
@@ -67,6 +69,19 @@ func (p *PEMCertPool) AppendCertsFromPEM(pemCerts []byte) (ok bool) {
 	return
 }
 
+// AppendCertsFromPEMFile adds certs from a file that contains concatenated PEM data.
+func (p *PEMCertPool) AppendCertsFromPEMFile(pemFile string) error {
+	pemData, err := ioutil.ReadFile(pemFile)
+	if err != nil {
+		return fmt.Errorf("failed to load PEM certs file: %v", err)
```
Comment (AlCutter) [inline examples/ct/pem_cert_pool.go]: Worth including the filename here (and below) for ease of debugging?

File: examples/ct/pem_cert_pool.go [hunk 2]
```diff
@@ -67,6 +69,19 @@ func (p *PEMCertPool) AppendCertsFromPEM(pemCerts []byte) (ok bool) {
...
+	}
+
+	if !p.AppendCertsFromPEM(pemData) {
```
Comment (AlCutter) [inline examples/ct/pem_cert_pool.go]: Feels like `AppendCertsFromPEM` should be returning and error (which gets passed on here).

File: integration/ct_integration_test.sh [hunk 3]
```diff
@@ -3,13 +3,22 @@ set -e
...
-TEST_TREE_ID=6962
 RPC_PORT=36962
 CT_PORT=6962
 
-echo "Provisioning test log (Tree ID: $TEST_TREE_ID) in database"
-${SCRIPTS_DIR}/wipelog.sh ${TEST_TREE_ID}
-${SCRIPTS_DIR}/createlog.sh ${TEST_TREE_ID}
+# Build config file with absolute paths
+CT_CFG=$(mktemp ${INTEGRATION_DIR}/ct-XXXXXX)
+sed "s!@TESTDATA@!$TESTDATA!" ./integration/ct_integration_test.cfg > ${CT_CFG}
```
Comment (AlCutter) [inline integration/ct_integration_test.sh]: `${TESTDATA}`

File: examples/ct/pem_cert_pool.go (hunk 1 above)
Comment (daviddrysdale) [inline examples/ct/pem_cert_pool.go]: `ioutil.ReadFile` will include the filename in its error already.

File: integration/ct_integration_test.sh (hunk 3 above)
Comment (daviddrysdale) [inline integration/ct_integration_test.sh]: Done.

File: examples/ct/pem_cert_pool.go (hunk 2 above)
Comment (AlCutter) [inline examples/ct/pem_cert_pool.go]: ok (pop a TODO?)

**Pull request merged**